test/ssl_test.c

```diff
@@ -196,20 +196,17 @@ static int check_nid(const char *name, int expected_nid, int nid)
 
 static void print_ca_names(STACK_OF(X509_NAME) *names)
 {
-    BIO *err;
     int i;
 
     if (names == NULL || sk_X509_NAME_num(names) == 0) {
-        fprintf(stderr, "    <empty>\n");
+        TEST_note("    <empty>");
         return;
     }
-    err = BIO_new_fp(stderr, BIO_NOCLOSE);
     for (i = 0; i < sk_X509_NAME_num(names); i++) {
-        X509_NAME_print_ex(err, sk_X509_NAME_value(names, i), 4,
+        X509_NAME_print_ex(bio_err, sk_X509_NAME_value(names, i), 4,
                            XN_FLAG_ONELINE);
-        BIO_puts(err, "\n");
+        BIO_puts(bio_err, "\n");
     }
-    BIO_free(err);
```

Comment:
The only function that needs to be able to print without a `#` is `run_tests` in `test/testutil/driverr.c`, so that one would need to be able to print "raw" lines.  Anything else should write indented lines starting with `#`.

Furthermore, you'll notice that `run_tests` writes on stdout while all others write to stderr.

So very simply put, I'd add that filter on top of `bio_err` only.  The trick is to make sure it treats each line separately.  `crypto/bio/bf_lbuf.c` might be worth having a look at.